clr/commands.py

```diff
@@ -41,27 +38,36 @@ def _load_namespace(key):
             return ErrorLoadingNamespace(key, error)
         instance = module.COMMANDS
     descr = instance.descr
-    longdescr = getattr(instance, 'longdescr', descr)
+    # Prefer doc string, otherwise explicit .longdescr, otherwise .descr
+    longdescr = inspect.getdoc(instance) or getattr(instance, 'longdescr', descr)
     command_callables = {
         attribute_name[4:]: getattr(instance, attribute_name)
         for attribute_name in dir(instance)
         if attribute_name.startswith('cmd_')
     }
-    command_specs = {
-        command_name: get_command_spec(command_callable)
-        for command_name, command_callable in command_callables.items()
-    }
+    # Build CommandSpecs for each command. These contain metadata about the
+    # command and its args. These are kept in a seperate dataclass from the
+    # callables because CommandSpec's are pickle-able and cached to disk.
+    command_specs = {}
+    for command_name, command_callable in command_callables.items():
+        docstr = inspect.getdoc(command_callable)
+        if docstr is None: docstr = ''
+        command_specs[command_name] = CommandSpec(
+            docstr,
+            Signature.from_callable(command_callable))
+
     return Namespace(
-        descr, longdescr, command_specs, command_callables, instance)
+        key, descr, longdescr, command_specs, command_callables, instance)
 
 def get_namespace(namespace_key):
-    """Lazily load and return the namespace"""
+    """Lazily load and return a namespace"""
     global __NAMESPACES
     if namespace_key not in __NAMESPACES:
         __NAMESPACES[namespace_key] = _load_namespace(namespace_key)
     return __NAMESPACES[namespace_key]
 
 def _get_close_matches(query, options):
+    """Utility function for making suggests in command not found messages."""
```

Comment:
Hm, I'm not really parsing what this description means. Is it making suggestions? What's a "command not found" message?